experimental/GModule/GModule.jl

```diff
@@ -112,7 +112,12 @@ case that `S` is a number field and `R` is the ring of integers in `S`,
 for example `R = ZZ` and `S = QQ`)
 """
 function extension_of_scalars(M::GModule, phi::Map)
-  error("not yet ...")
+  @assert domain(phi) == base_ring(M)
+
+  d = dim(M)
+  F = free_module(codomain(phi), d)
+
+  return GModule(F, group(M), [hom(F, F, map_entries(phi, matrix(x))) for x in M.ac])
 end
```

Comment:
There should be test cases for all these functions. Ideally also `jldoctest` examples in their docstrings. And the docstrings should be inserted into the manual.